app/docs/0.10.x/proxy.md

```diff
@@ -0,0 +1,759 @@
+port.
+- `Upstream service`: Refers to your own API/service setting behind Kong,
+where requests are forwarded to.
+
+[Back to TOC](#table-of-contents)
+
+### Overview
+
+From a high level perspective, Kong will listen for HTTP traffic on its
+configured proxy port (`8000` by default), recognize which upstream service is
+being requested, run the configured plugins for that API, and forward the HTTP 
+request upstream to your own API or service. Kong can also run plugins once the 
+response from your upstream service is received, but that topic is not as
+relevant to this guide.
+
+When a client makes a request to the proxy port, Kong will decide where to route
+(or forward) that incoming request (that is, to which of your upstream 
+services). Kong achieves this through the APIs you configured (via the Admin 
+API). You can configure APIs with various properties, but the three relevant 
+ones for routing incoming traffic are `hosts`, `uris` and `methods`.
+
+If Kong cannot determine to which API a given request should be routed to, it
+will response with:
+
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+Server: kong/<x.x.x>
+
+{
+    "message": "no API found with those values"
+}
+```
+
+### Reminder: how to add an API to Kong
+
+Before goign any further, let's take a few moments to make sure you know how to
+add an API to Kong.
+
+As explained in the [Adding your API][adding-your-api] quickstart guide, Kong is
+configured via its internal [Admin API][API] running by default on port `8001`. 
+Adding an API to Kong is as easy as sending an HTTP request:
+
+```bash
+$ curl -i -X POST http://localhost:8001/apis/ \
+    -d 'name=my-api' \
+    -d 'upstream_url=http://my-api.com' \
+    -d 'hosts=my-api.com' \
+    -d 'uris=/my-api' \
+    -d 'methods=GET,HEAD'
+HTTP/1.1 201 Created
+...
+```
+
+This request instructs Kong to register an API named "my-api", reachable at
+"http://my-api.com". It also specifies various routing properties. In reality,
+**only one of** `hosts`, `uris` and `methods` is actually required. More on this
+later.
+
+Adding such an API would mean that you configured Kong to proxy all incoming 
+requests matching `hosts`, `uris`, and `methods` to `http://my-api.com`.
+Kong being a transparent proxy, it will forward the request to your upstream
+service - practically - untouched (exception made of various headers such as
+`Connection`, or in case a plugin performs some transformations to it).
+
+[Back to TOC](#table-of-contents)
+
+### Routing capabilities
+
+Let's now investigate in details how Kong matches a request to the configured
+`hosts`, `uris` and `methods` properties of your API:
+
+- All of those fields are **optional**, but at least **one of them** must be 
+specified.
+- For a request to match an API, all of configured fields **must** be satisfied
+(present in the client HTTP request).
+- All of those fields accept one or more values. A field only needs one of 
```

Comment:
`All of those` should be `All of these` as we are currently in the context of these fields.